packages/dds/tree/src/feature-libraries/editable-tree/utilities.ts

```diff
@@ -86,3 +87,20 @@ export function adaptWithProxy<From extends object, To extends object>(
 export function getArrayOwnKeys(length: number): string[] {
     return Object.getOwnPropertyNames(Array.from(Array(length)));
 }
+
+const pathDelimiter = "/";
+
+function stringifyPath(path: UpPath, childPath: ETreeNodePath): ETreeNodePath {
+    const fieldName = typeof path.parentField === "string" ? path.parentField : keyFromSymbol(path.parentField);
+    return brand(`${fieldName}[${path.parentIndex}]${childPath}`);
```

Comment:
currently we have no rules on whats allowed in a key. SO assuming field key do not contain `[` `]` or `\` are new assumptions. We should either handle those cases, or clearly document they are not supposed and check for them.

Reply:
I am working on getting rid of this way of caching like e.g. you've suggested above. If I'll find no other way, this for sure has to be discussed and properly documented.